content/engine/release-notes/25.0.md

```diff
@@ -30,6 +30,7 @@ For a full list of pull requests and changes in this release, refer to the relev
 
 ### New
 
+- The daemon now uses systemd's default `LimitNOFILE` which on older systems can be a low value, which may limit the number of containers that can be run. Set `LimitNOFILE=1048576` to get the previous behavior. [moby/moby#45534](https://github.com/moby/moby/pull/45534)
```

Comment:
Perhaps we should mention; that previously we set both the soft _and_ hard limit to `infinity`, which caused issues with recent versions of systemd v240 and up, which raised that from `1048576 ` to `1073741816`, causing some software that adjust their behavior based on ulimits to consume a lot of memory (see  https://github.com/containerd/containerd/issues/6707#issuecomment-1148727751).

The new setting makes containers use systemd's default (`1024:524288` on current versions of systemd), which makes containers have identical behavior as when running on the host, but  may cause some software that makes incorrect assumptions based on the soft-limit to misbehave (see e.g. envoy https://github.com/containerd/containerd/pull/8924#issuecomment-1867916971)


w.r.t. "on older systems" part; I recall we update our packages for CentOS 7, where this was a problem; https://github.com/docker/docker-ce-packaging/pull/968, so for "most" use-cases, things should be fine, but _perhaps_ for debian buster that may be an issue (looks like we still build packages for buster, but don't advertise it).

Reply:
Hm.. and just when I wrote that, I'm forgetting that this may all only apply to containers that are running as part of `docker build` with BuildKit, because this setting applies to the docker daeemon (and child processes), but containers we start still use containerd's setting, which has not changed yet for the version of containerd (1.6, 1.7) we use; https://github.com/docker/cli/issues/4807

Reply:
Hmf... so some help may be needed what (and how) to describe here 😬 🙈

Reply:
I tried.. what do you think about it now?